packages/sdk/src/dataProtector/sharing/addToCollection.ts

```diff
@@ -0,0 +1,166 @@
+  // Example in zod: https://zod.dev/?id=functions
+  // const vonStatusUpdate: string = fnSchema().label('onStatusUpdate').validateSync(onStatusUpdate);
+
+  const vProtectedDataAddress = addressSchema()
+    .required()
+    .label('protectedDataAddress')
+    .validateSync(existingProtectedDataAddress);
+  const vCollectionId = positiveNumberSchema()
+    .required()
+    .label('collectionId')
+    .validateSync(collectionId);
+
+  const userAddress = (await iexec.wallet.getAddress()).toLowerCase();
+
+  const protectedData = await checkAndGetProtectedData({
+    graphQLClient,
+    protectedDataAddress: vProtectedDataAddress,
+    userAddress,
+  });
+
+  await checkCollection({
+    iexec,
+    sharingContractAddress,
+    collectionId: vCollectionId,
+    userAddress,
+  });
+
+  onStatusUpdate?.({
+    title: 'Give ownership to the collection smart-contract',
+    isDone: false,
+  });
+  // Approve collection SC to change the owner of my protected data in the registry SC
+  await approveCollectionContract({
+    iexec,
+    protectedData,
+    sharingContractAddress,
+  });
+  onStatusUpdate?.({
+    title: 'Give ownership to the collection smart-contract',
+    isDone: true,
+  });
+
+  onStatusUpdate?.({
+    title: 'Add protected data to your collection',
+    isDone: false,
+  });
+  await addProtectedDataToCollection({
+    iexec,
+    sharingContractAddress,
+    collectionId: vCollectionId,
+    protectedDataAddress: vProtectedDataAddress,
+  });
+  onStatusUpdate?.({
+    title: 'Add protected data to your collection',
+    isDone: true,
+  });
+};
+
+async function checkAndGetProtectedData({
+  graphQLClient,
+  protectedDataAddress,
+  userAddress,
+}: {
+  graphQLClient: GraphQLClient;
+  protectedDataAddress: Address;
+  userAddress: Address;
+}) {
+  const protectedData = await getProtectedDataById({
+    graphQLClient,
+    protectedDataAddress,
+  });
+
+  if (!protectedData) {
+    throw new ErrorWithData(
+      'This protected data does not exist in the subgraph.',
+      { protectedDataAddress }
+    );
+  }
+
+  if (protectedData.owner.id !== userAddress) {
+    throw new ErrorWithData('This protected data is not owned by the user.', {
+      protectedDataAddress,
+      userAddress,
+    });
+  }
+
+  return protectedData;
+}
+
+async function checkCollection({
+  iexec,
+  collectionId,
+  userAddress,
+  sharingContractAddress,
+}: {
+  iexec: IExec;
+  collectionId: number;
+  userAddress: Address;
+  sharingContractAddress: AddressOrENS;
+}) {
+  const { provider, signer } = await iexec.config.resolveContractsClient();
+
+  const { collectionContract } = await getCollectionContract({
+    provider,
+    sharingContractAddress,
+  });
+
+  const ownerAddress: Address = await (
+    collectionContract.connect(signer) as Contract
+  )
+    .ownerOf(collectionId)
+    .catch(() => {
+      // Do nothing
+      // An error means that the collection does not exist
```

Comment:
to do in next PR 🤷🏻‍♂️ : 
```
} catch (error: any) {
    // catch the revert custom error : error ERC721NonexistentToken(uint256 tokenId);
    error = collectionContract.interface.decodeErrorResult(
      "ERC721NonexistentToken",
      error.data.data
    )[0];
    console.log("decoded error", error);```

Reply:
I've added that but commented for now (`error.data` is still undefined...)